### PR TITLE
Add swrast and null information to rendering architecture

### PR DESCRIPTION
**Pull request by keirthana**

### CODE REVIEW

File: exp/rendering-architecture.md [hunk 1]
```diff
@@ -15,6 +17,16 @@ For AMD and Intel GPUs, Anbox Cloud uses the [Mesa driver](https://www.mesa3d.or
 
 See [Component versions](https://anbox-cloud.io/docs/component-versions) to refer to the actual version supported for any particular Anbox Cloud release.
 
+# Supported platforms
+
+For rendering, you can use the `swrast` or the `null` platforms depending on your requirements.
+
+`swrast` is a software rasterization platform, which is a rendering implementation of the Mesa driver with support for LLVMpipe. It can be utilised for use cases that require a visual output without a GPU. When you set `swrast` with LLVMpipe as your backend in the `session.yaml` file, the rendering pipeline uses this platform irrespective of any available GPUs. To know more about this implementation, see [LLVMpipe](https://docs.mesa3d.org/drivers/llvmpipe.html).
```
Comment (adglkh) [inline exp/rendering-architecture.md:24]: `When you set `swrast` with LLVMpipe as your backend in the `session.yaml` file,`
This involves the technical details which we don't want our end user to dive into. maybe the following fits better.
```
When you launch an Anbox container with the `swrast` platform via:
    amc launch -p swrast <app_id>
```

`the rendering pipeline uses this platform irrespective of any available GPUs.`
->
The rendering pipe for the swrast or null platform has nothing different than the one for the webrtc platform with NVIDIA GPU support except for the fact that it's irrespective of any available GPUs

File: exp/rendering-architecture.md [hunk 2]
```diff
@@ -15,6 +17,16 @@ For AMD and Intel GPUs, Anbox Cloud uses the [Mesa driver](https://www.mesa3d.or
...
+
+`null` is a backend renderer of the [Almost Native Graphics Layer Engine (ANGLE)](https://chromium.googlesource.com/angle/angle) and can be used when you do not need a graphic output, such as, automation testing.
```
Comment (adglkh) [inline exp/rendering-architecture.md:26]: `null ` is an OpenGL headless platform, which makes use of the `null` rendering backend of the [Almost Native Graphics Layer Engine (ANGLE)](https://chromium.googlesource.com/angle/angle)...

**Review by adglkh — CHANGES_REQUESTED** [review]

File: exp/rendering-architecture.md [hunk 3]
```diff
@@ -15,6 +17,16 @@ For AMD and Intel GPUs, Anbox Cloud uses the [Mesa driver](https://www.mesa3d.or
...
 
+# Supported platforms
+
+For rendering, you can use the `swrast` or the `null` platforms depending on your requirements.
+
+`swrast` is a software rasterization platform, which is a rendering implementation of the Mesa driver with support for LLVMpipe. It can be utilised for use cases that require a visual output without a GPU. When you launch an Anbox container with the `swrast` platform using the `amc launch -p swrast <app_id>` command, the rendering pipeline uses this platform irrespective of any available GPUs. The rendering pipe for the `swrast` or `null` platform is not different than the one for the `webrtc` platform with NVIDIA GPU support except for the fact that it's irrespective of any available GPUs. To know more about this implementation, see [LLVMpipe](https://docs.mesa3d.org/drivers/llvmpipe.html).
+
+`null` is an OpenGL headless platform that makes use of the rendering backend of the [Almost Native Graphics Layer Engine (ANGLE)](https://chromium.googlesource.com/angle/angle) and can be used when you do not need a graphic output, such as, automation testing.
```
Comment (morphis) [inline exp/rendering-architecture.md:26]: I feel this doesn't emphazize enough that the `null` platform does not produce any pixel output or even perform software rendering. We should also highlight that it has very low overhead on the CPU is with that perfect for all use cases where no graphical output is necessary.

**Review by morphis — CHANGES_REQUESTED** [review]

**Review by morphis — APPROVED** [review]
Thanks, LGTM!

File: exp/rendering-architecture.md [hunk 4]
```diff
@@ -15,6 +17,16 @@ For AMD and Intel GPUs, Anbox Cloud uses the [Mesa driver](https://www.mesa3d.or
 
 See [Component versions](https://anbox-cloud.io/docs/component-versions) to refer to the actual version supported for any particular Anbox Cloud release.
 
+# Supported platforms
+
+For rendering, you can use the `swrast` or the `null` platforms depending on your requirements.
+
+`swrast` is a software rasterization platform, which is a rendering implementation of the Mesa driver with support for LLVMpipe. It can be utilised for use cases that require a visual output without a GPU. When you launch an Anbox container with the `swrast` platform using the `amc launch -p swrast <app_id>` command, the rendering pipeline uses this platform irrespective of any available GPUs. The rendering pipe for the `swrast` or `null` platform is not different than the one for the `webrtc` platform with NVIDIA GPU support except for the fact that it's irrespective of any available GPUs. To know more about this implementation, see [LLVMpipe](https://docs.mesa3d.org/drivers/llvmpipe.html).
```
Comment (adglkh) [inline exp/rendering-architecture.md:24]: ` the rendering pipeline uses this platform irrespective of any available GPUs`
Can we drop this sentence as the following one cover the point already?

Comment (keirthana) [reply exp/rendering-architecture.md:24]: @adglkh 

> When you launch an Anbox container with the `swrast` platform using the `amc launch -p swrast <app_id>` command, the rendering pipeline uses this platform irrespective of any available GPUs.

This statement also informs the user they can launch a container with the particular command. Are you fine with dropping this entire statement? Or just the last part that talks about the pipeline irrespective of GPUs?

Comment (adglkh) [reply exp/rendering-architecture.md:24]: I feel these two sentences are a bit repetitive as they're both explaining the point that the rendering pipeline uses this platform irrespective of any available GPUs. I vote for dropping this entire statement and only keeping `The rendering pipe for the `swrast` or `null` platform...`
what do you think?

Comment (keirthana) [reply exp/rendering-architecture.md]: @adglkh Agree, done.

**Review by adglkh — APPROVED** [review]
LGTM

**Pull request merged**